### PR TITLE
Chore: Update prometheus, thanos and promql-engine

### PR DESCRIPTION
**Pull request by harry671003**
**What this PR does**:
- Update prometheus to: e9dec5fc537b
- Update thanos to: 990a60b72647
- Update promql-engine to: cdbd6dfed36b

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/cortexproject/cortex
 
-go 1.22
+go 1.22.0
+
+toolchain go1.22.5
```
Comment (yeya24) [inline go.mod:5]: Is there a way to avoid this? If not, then we can live it...

File: pkg/ruler/user2/namespace [hunk 2]
```diff
@@ -0,0 +1,4 @@
+groups:
```
Comment (CharlieTLe) [inline pkg/ruler/user2/namespace:1]: Should these files be checked in?

Comment (yeya24) [inline pkg/ruler/user2/namespace:1]: I think those files are accidently committed and they are part of the ruler unit tests.
Let's remove them. Maybe we should change the unit tests to use /tmp dir

File: go.mod [hunk 3]
```diff
@@ -46,14 +48,14 @@ require (
...
+	github.com/prometheus/prometheus v0.53.2-0.20240718123124-e9dec5fc537b
 	github.com/segmentio/fasthash v1.0.3
 	github.com/sony/gobreaker v1.0.0
 	github.com/spf13/afero v1.11.0
 	github.com/stretchr/testify v1.9.0
 	github.com/thanos-io/objstore v0.0.0-20240622095743-1afe5d4bc3cd
-	github.com/thanos-io/promql-engine v0.0.0-20240515161521-93aa311933cf
-	github.com/thanos-io/thanos v0.36.0-rc.0
+	github.com/thanos-io/promql-engine v0.0.0-20240718195911-cdbd6dfed36b
+	github.com/thanos-io/thanos v0.35.2-0.20240722172812-990a60b72647
```
Comment (CharlieTLe) [inline go.mod:58]: Seems like we downgraded from thanos version `v0.36.0-rc.0` to `v0.35.2-0`?

**Review by CharlieTLe — COMMENTED** [review]
There seems to be unrelated changes like the addition of `storage.LabelHints` in some function calls. Are these related to the dependency updates?

File: go.mod (hunk 1 above)
Comment (harry671003) [reply go.mod:5]: I'm not familiar with this.
Thanos also had the same:
- https://github.com/thanos-io/promql-engine/blob/main/go.mod#L5
- https://github.com/thanos-io/thanos/blob/main/go.mod#L5

File: go.mod (hunk 3 above)
Comment (harry671003) [reply go.mod:58]: 990a60b72647 is newer. 


v0.36.0-rc.0 is from one month ago.

**Comment by harry671003** [comment]
> There seems to be unrelated changes like the addition of storage.LabelHints in some function calls. Are these related to the dependency updates?

The interface for the `storage.Querier` was changed in https://github.com/prometheus/prometheus/pull/14109/

File: go.mod (hunk 1 above)
Comment (harry671003) [reply go.mod:5]: Related https://github.com/golang/go/issues/65847

Comment (yeya24) [reply go.mod:5]: Let's live with it then. I see Prometheus has the same config.

**Review by yeya24 — APPROVED** [review]
Thanks a lot

**Review by CharlieTLe — APPROVED** [review]
Thanks

**Pull request merged**